tests/e2e/fio_test.go

```diff
@@ -17,15 +17,78 @@ limitations under the License.
 package e2e
 
 import (
+	"fmt"
+	"io/ioutil"
+
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"github.com/xridge/kubestone/api/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	fioCrBaseDir    = samplesDir + "/fio"
+	e2eNamespaceFio = "kubestone-e2e-fio"
 )
 
 var _ = Describe("end to end test", func() {
-	Describe("for fio", func() {
-		// TODO: Add fio end-to-end test
-		It("Should be implemented", func() {
-			Expect(true).To(BeTrue())
+	Context("preparing namespace", func() {
+		_, _, err := run("kubectl create namespace " + e2eNamespaceFio)
+		It("should succeed", func() {
+			Expect(err).To(BeNil())
 		})
 	})
+
+	fioCrDirs := []string{fioCrBaseDir + "/base"}
+	fioOverlayDirs, err := ioutil.ReadDir(fioCrBaseDir + "/overlays")
+	if err != nil {
+		Fail("Didn't find any fio CRs under " + fioCrBaseDir)
+	}
+	for _, fioOverlayDir := range fioOverlayDirs {
+		fioCrDirs = append(fioCrDirs, fioCrBaseDir+"/overlays/"+fioOverlayDir.Name())
+	}
+
+	Describe("creating fio job from multiple CRs", func() {
+		for i, fioCrDir := range fioCrDirs {
+			crName := fmt.Sprintf("fio-sample-%d", i)
+
+			Context("when creating from cr", func() {
+				_, _, err := run(`bash -c "` +
+					"kustomize build " + fioCrDir + " | " +
+					"sed 's/fio-sample/" + crName + "/' | " +
+					"kubectl create -n " + e2eNamespaceFio + ` -f -"`)
+				It("should create fio-sample cr", func() {
+					Expect(err).To(BeNil())
+				})
+			})
+
+			Context("the created job", func() {
+				It("should finish in a pre-defined time", func() {
+					timeout := 120
```

Comment:
Can we use a smaller timeout here?

Reply:
No, this is needed for the `custom_jobs` example

Reply:
can we then change the custom_jobs?  we are already spending 5m on the system test path...

Reply:
done